hydromt/data_adapter/rasterdataset.py

```diff
@@ -307,8 +308,7 @@ def _resolve_paths(
             else:
                 os.environ["AWS_NO_SIGN_REQUEST"] = "NO"
 
-        # parse zoom level (raster only)
-        if len(self.zoom_levels) > 0:
+        if zoom_level is not None and "{zoom_level}" in self.path:
```

Comment:
isn't this going to trigger if the path contains basically just the digit of the zoom_level anywhere in the path? Not quite sure if there is a way around it right now, but something to think about at least.

Reply:
Oh also, I think you meant
```suggestion
        if zoom_level is not None and f"{zoom_level}" in self.path:
```
This checks for the literal string "{zoom_level}" and not the value

Reply:
It should actually check if a `"{zoom_level}"` is in the path, e.g., True in case of `"path/to/my/files/{zoom_level}/data.tif"` 
The value is then computed by `_parse_zoom_level` when a zoom_level is passed to `get_data`

Reply:
aaaah right gotcha, my bad